pages/steps/with-updates.page.tsx

```diff
@@ -0,0 +1,170 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useRef, useState } from 'react';
+
+import Box from '~components/box';
+import Button from '~components/button';
+import Header from '~components/header';
+import Steps from '~components/steps';
+
+import {
+  blockedStepsInteractive,
+  failedStepsInteractive,
+  failedStepsWithRetryButtonInteractive,
+  initialStepsInteractive,
+  loadingSteps2Interactive,
+  loadingSteps3Interactive,
+  loadingStepsInteractive,
+  successfulStepsInteractive,
+} from './permutations-utils';
+
+export default function StepsPermutationsWithUpdates() {
+  const [stepIndex1, setStepIndex1] = useState(0);
+  const timeoutRef1 = useRef<ReturnType<typeof setTimeout>>();
+  const stepsExecution1 = [
+    initialStepsInteractive,
+    loadingStepsInteractive,
+    loadingSteps2Interactive,
+    loadingSteps3Interactive,
+    successfulStepsInteractive,
+  ];
+
+  const [stepIndex2, setStepIndex2] = useState(0);
```

Comment:
This can probably be refactored with a component that manages its own internal state independently, instead of managing 4 almost-identical states in a single component.

Reply:
That is true, but is it worth it at this point? :) It's just an example page

Reply:
Let's call this non-blocking but a suggestion. Always good to keep things simple for the next one who might work on it.